src/test/java/com/google/jenkins/plugins/k8sengine/KubernetesEnginePublisherTest.java

```diff
@@ -84,20 +85,61 @@ public void testDoFillZoneItemsWithValidArguments() {
   }
 
   @Test
-  public void testDoFillZoneItemsWithValidArgumentsNoZones() {
+  public void testDoFillZoneItemsEmptyWithValidArgumentsNoZones() {
     ListBoxModel zones = descriptor.doFillZoneItems(jenkins, TEST_PROJECT_ID, TEST_CREDENTIALS_ID);
     testZoneEmptyResult(zones);
   }
 
   @Test
-  public void testDoFillZoneItemsIOException() {
+  public void testDoFillZoneItemsErrorMessageWithIOException() {
     ListBoxModel zones = descriptor.doFillZoneItems(jenkins, ERROR_PROJECT_ID, TEST_CREDENTIALS_ID);
     assertNotNull(zones);
     assertEquals(1, zones.size());
     assertEquals(Messages.KubernetesEnginePublisher_ZoneFillError(), zones.get(0).name);
     assertTrue(Strings.isNullOrEmpty(zones.get(0).value));
   }
 
+  @Test
+  public void testDoCheckZoneMessageWithEmptyZone() {
+    FormValidation result =
+        descriptor.doCheckZone(jenkins, null, TEST_PROJECT_ID, TEST_CREDENTIALS_ID);
+    assertNotNull(result);
+    assertEquals(Messages.KubernetesEnginePublisher_ZoneRequired(), result.getMessage());
+  }
+
+  @Test
+  public void testDoCheckZoneMessageWithEmptyProjectOrCredentialsId() {
+    FormValidation result = descriptor.doCheckZone(jenkins, TEST_ZONE_A, null, TEST_CREDENTIALS_ID);
+    assertNotNull(result);
+    assertEquals(
+        Messages.KubernetesEnginePublisher_ZoneProjectIdCredentialRequired(), result.getMessage());
+    result = descriptor.doCheckZone(jenkins, TEST_ZONE_A, TEST_PROJECT_ID, null);
+    assertNotNull(result);
+    assertEquals(
```

Comment:
Suggest splitting this into a separate test case. Each case should ideally contain a single logical test. Same feedback for the other test cases as well.

Reply:
Just to clarify, that means the tests should have one single input case to one single expected outcome?

Reply:
Let me know what you think of how I split the tests. I separated the empty cases for each parameter into their own tests, and used different tests for different contexts (i.e. no zones available vs. wrong zone entered)

Reply:
LGTM